theano/gpuarray/extra_ops.py

```diff
@@ -450,12 +450,13 @@ def c_support_code_struct(self, node, nodename):
         """ % locals()
         return super(GpuCumsum, self).c_support_code_struct(node, nodename) + code
 
-
+@register_opt('fast_compile')
 @op_lifter([CumsumOp])
-def use_gpu_cumsumop(node, ctx_name):
-    if node.inputs[0].dtype == 'float32':
-        axis = node.op.axis
-        x = node.inputs[0]
+@register_opt2([CumsumOp], 'fast_compile')
```

Comment:
We need to register this in register_opt2. But this opt needed more change then the previous one. I think the diff I gave you should fix it. Otherwise, make a breakpoint here and check why it fail.

The error you got was because this opt was raising a TypeError, but for another reason then the number of parameter.